### DISCUSSION TITLE
Hubspot moving to Private Apps

### DISCUSSION THREAD

**Discussion by ZJensenDev** [solspace/craft-freeform#421] [Questions & Answers] [answered]
Hubspot is moving all integrations to private apps rather than through the API, are there plans to update the integration through Freeform? They sent out a November deadline to get everything moved over.

**Comment by kjmartens** [comment, marked as answer]
Hi @ZJensenDev,

Yes, we are aware of this and currently in the process of updating the HubSpot API integration. It'll be ready sometime in the next week. 🙂

**Reply by ZJensenDev** [reply]
Thanks so much!

**Reply by mylesthedev** [reply]
Was there an ETA for this?

**Comment by kjmartens** [comment]
This has now been resolved in Freeform 3.13.18 and 4.0.1. 🙂 

You can refer to the special update instructions here for existing users:

- [Freeform 3.x](https://docs.solspace.com/craft/freeform/v3/integrations/crm/hubspot.html#_2022-private-key-update-instructions)
- [Freeform 4.x](https://docs.solspace.com/craft/freeform/v4/integrations/crm/hubspot.html#_2022-private-key-update-instructions)

**Reply by mylesthedev** [reply]
🔥
